airflow-core/src/airflow/api_fastapi/common/parameters.py

```diff
@@ -185,6 +185,64 @@ def depends(cls, *args: Any, **kwargs: Any) -> Self:
         raise NotImplementedError("Use search_param_factory instead , depends is not implemented.")
 
 
+class QueryTITaskGroupDisplayNamePattern(BaseParam[str]):
+    """Task group display name pattern filter - returns all tasks in matching groups."""
+
+    def __init__(self, dag=None, skip_none: bool = True):
+        super().__init__(skip_none=skip_none)
+        self.dag = dag
```

Comment:
Would it be better to provide a setter for `dag` attribute and add validation at `to_orm` to check whether the `dag` attribute is set? Just in case we need the same query class in some other place and having a explicit setter might be more readable.

```suggestion
        self.dag: None | SerializedDAG = dag
```